kv/memberlist/tcp_transport.go

```diff
@@ -426,7 +432,15 @@ func (t *TCPTransport) getAdvertisedAddr() string {
 func (t *TCPTransport) WriteTo(b []byte, addr string) (time.Time, error) {
 	t.sentPackets.Inc()
 	t.sentPacketsBytes.Add(float64(len(b)))
+	t.writeCh <- struct{}{}
+	go func() {
+		defer func() { <-t.writeCh }()
+		t.writeToAsync(b, addr)
+	}()
```

Comment:
Instead of starting new goroutine for each write, shall we have a pool of writing goroutines, and pass them messages to write via channel?

Reply:
I've updated the PR to have that